cloudformation/tag-manager.yaml

```diff
@@ -284,6 +282,69 @@ Resources:
         Value: flexible
       - Key: App
         Value: tag-manager
+  TagManagerCloudfrontCachePolicy:
+    Type: "AWS::CloudFront::CachePolicy"
+    Properties:
+      CachePolicyConfig:
+        DefaultTTL: 86400
+        MaxTTL: 31536000
+        MinTTL: 0
+        Name: !Sub "tag-manager-cache-policy-${Stage}"
+        ParametersInCacheKeyAndForwardedToOrigin:
+          CookiesConfig:
+            CookieBehavior: "all"
+          EnableAcceptEncodingBrotli: false
+          EnableAcceptEncodingGzip: false
+          HeadersConfig:
+            HeaderBehavior: "whitelist"
+            Headers:
+              - "Host"
+              - "Origin"
+              - "Access-Control-Request-Headers"
+              - "Access-Control-Request-Method"
+              - "X-Gu-Tools-HMAC-Token"
+              - "X-Gu-Tools-HMAC-Date"
+              - "X-Gu-Tools-Service-Name"
+          QueryStringsConfig:
+            QueryStringBehavior: "all"
+  TagManagerCloudFrontDistribution:
+    Type: "AWS::CloudFront::Distribution"
+    Properties:
+      DistributionConfig:
+        Aliases:
+          - !FindInMap [Config, !Ref 'Stage', DNSName]
+        DefaultCacheBehavior:
+          AllowedMethods:
+            - "GET"
+            - "HEAD"
+            - "OPTIONS"
+          CachePolicyId: !Ref 'TagManagerCloudfrontCachePolicy'
+          Compress: true
+          TargetOriginId: !Ref 'TagManagerLoadBalancer'
+          ViewerProtocolPolicy: "allow-all"
```

Comment:
I think this should be redirect-to-https

Reply:
Now done.